### PR TITLE
feat: Add create/load speaker profile for user

### PR DESCRIPTION
**Pull request by anhanh11001**
Details:
- Create/Load speaker profile for proposal
- I'll finish the rest of #1811 (implementing create proposal after this PR is finished and merged as this is already a large PR)

Fixes: #1811

Screenshots for the change:
       

ezgif-4-94cf806dc384 
 ezgif-4-6315722607c0

### CODE REVIEW

**Comment by anhanh11001** [comment]
@nikit19 Yes, I understand it, and both current WIP PR I'm sending take me more time then I expected, but I'll make sure I finish them as soon as possible, hopefully around today evening. I sent them to let the mentors and other contributors know that I'm still making some progress

**Comment by anhanh11001** [comment]
@nikit19 Do you know how to make a filter in making GET request. I'm trying to take a speaker under event and user by this:
```
@GET("users/{userId}/speakers?include=event,user&fields[event]=id&fields[user]=id")
    fun getSpeakerUnderUserAndEvent
        (@Path("userId") userId: Long, @Query("event", encoded = true) eventId: EventId): Single<Speaker>
```
or
```
@GET("users/{userId}/speakers?include=event,user&fields[event]=id&fields[user]=id")
    fun getSpeakerUnderUserAndEvent
        (@Path("userId") userId: Long, @Query("event", encoded = true) eventId: Long): Single<Speaker>
```
But both of this return a list of speakers under user without filtering the event ID given.

**Comment by iamareebjamal** [comment]
There's no filter applied and the query itself is wrong. Why are you querying `users/{user_id}`?

**Comment by anhanh11001** [comment]
In the frontend, the information about my speaker is stored in each event when I try to send a proposal. If not, then I would have to create a new speaker. 



I'm trying to do the same thing but I don't know how to save the current speaker detail. So I was trying to create a speaker under a user and event, save that in the server and then query it to check whether I have an existing user or not.

I query `users/{user_id}` as I follow this from the documentation. This does return a list of speakers
Capture

**Comment by iamareebjamal** [comment]
But you haven't applied any filter

**Comment by anhanh11001** [comment]
So putting this `@Query("event", encoded = true) eventId: EventId` in the function is not correct way to filter.. . So I'll spend time on reading about the filter part and I will update if I still have problems

**Comment by iamareebjamal** [comment]
`users/{userId}/speakers?include=event,user&fields[event]=id&fields[user]=id`

There is no filter in the query. Adding event in the query will just make it 

`users/33/speakers?include=event,user&fields[event]=id&fields[user]=id&event=234`

Also you are using encoded EventId and not an int, which will result in God knows what

How filters work if shown in Search section of the app. Please take a look at that

**Comment by anhanh11001** [comment]
Updated: Description above

File: app/src/main/java/org/fossasia/openevent/general/speakers/Speaker.kt [hunk 1]
```diff
@@ -15,24 +19,30 @@ data class Speaker(
...
+    val facebook: String? = null,
+    val linkedin: String? = null,
+    val github: String? = null,
+    val isFeatured: Boolean = false,
+    @ColumnInfo(index = true)
+    @Relationship("event")
+    var event: EventId? = null,
+    @ColumnInfo(index = true)
+    @Relationship("user")
+    var user: UserId? = null
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/speakers/Speaker.kt]: These all should be value not variables, right?

**Comment by anhanh11001** [comment]
Updated, I have used progress dialog, it looks better; enable email editing (at first I thought the frontend set is as disabled); turned variables to values

File: app/src/main/java/org/fossasia/openevent/general/speakercall/SpeakersCallViewModel.kt [hunk 2]
```diff
@@ -6,30 +6,91 @@ import androidx.lifecycle.ViewModel
...
+                |       'op':'has',
+                |       'val': {
+                |           'name': 'identifier',
+                |           'op': 'eq',
+                |           'val': '$eventIdentifier'
+                |       }
+                |    }, {
+                |       'name':'email',
+                |       'op':'eq',
+                |       'val':'${user.email}'
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/speakercall/SpeakersCallViewModel.kt]: Why is the email thingy needed?

**Comment by iamareebjamal** [comment]
I think you can remove the email from the filter. Rest is fine

File: app/src/main/java/org/fossasia/openevent/general/speakercall/SpeakersCallViewModel.kt (hunk 2 above)
Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/speakercall/SpeakersCallViewModel.kt]: I followed the implementation in the frontend. It seems like the server allow a user to create multiple speaker's profiles, but only allow one email per profile. 

dfs

**Pull request merged**